django/forms/fields.py

```diff
@@ -306,9 +307,9 @@ def to_python(self, value):
 
     def validate(self, value):
         super().validate(value)
-
-        # Check for NaN (which is the only thing not equal to itself) and +/- infinity
-        if value != value or value in (Decimal('Inf'), Decimal('-Inf')):
+        if value in self.empty_values:
+            return
+        if math.isnan(float(value)) or math.isinf(float(value)):
```

Comment:
It seems that `math.isfinite()` checks both for `NaN` and `Inf`.
I don't think cast to `float` is needed here.